File: posts/2023053103_retro_platformer_3/sketch.js
// Retro Platformer Stage 2
// Mit P5.play und planck.js
// Assets: Pixel Adventure von Pixel Frog: https://pixelfrog-assets.itch.io/pixel-adventure-1

// Konstanten
const winwidth = 640;
const winheight = 240;
const ninjaSpeed = 1.0;

// Globale Variablen
let ninja;
let ninjaIdleAni;
let ninjaWalkAni;
let idleAni;
let walkAni;


// Funktionen
function displayGround() {
  push();
  noStroke();
  fill(150, 63, 62);  // Rotbraun
  rect(0, height - 50, width, height);
  pop();
}

// Hauptprogramm
function preload() {
  ninjaIdleAni = loadAni("data/ninjafrog/idle/ninjafrog_idle_00.png", 10);
  ninjaWalkAni = loadAni("data/ninjafrog/walk/ninjafrog_walk_00.png", 11);
}

function setup() {
  myCanvas = createCanvas(winwidth, winheight);
  myCanvas.parent("mysketch");
  ninja = new Sprite(width/2, 174);
  walkAni = ninja.addAni("walk", ninjaWalkAni);
  idleAni = ninja.addAni("idle", ninjaIdleAni);
  ninja.mirror.x = false;
  ninja.speed = 0;
  } // end setup()

function draw() {
  background(215, 189, 156);  // Ocker 
  displayGround();

  if (kb.pressing("left")) {
    ninja.mirror.x = true;
    ninja.ani = "walk";
    if (ninja.x > 16) {
      ninja.direction = 180;
      ninja.speed = ninjaSpeed;
    } else {
      ninja.speed = 0;
    }
  } else if (kb.pressing("right")) {
    ninja.mirror.x = false;
    ninja.ani = "walk";
    if (ninja.x < width - 16) {
      ninja.direction = 0;
      ninja.speed = ninjaSpeed;
    } else {
    ninja.speed = 0;
    }
  } else {
    ninja.ani = "idle";
    ninja.speed = 0;
  }
} // end draw()

// End Hauptprogramm

File: docs/posts/2025062901_proseco_first_steps/proseco/sketch.py
from proceso import Sketch
from random import randint

p5 = Sketch()

def setup():
    p5.create_canvas(640, 400)
    p5.background(49, 197, 244)
    p5.fill(146, 82, 161)
    p5.rect(40, 40, p5.width - 80, p5.height - 80)
    p5.fill(randint(30, 220), randint(30, 220), randint(30, 220), 100)


def draw():
    p5.circle(p5.mouse_x, p5.mouse_y, 20)

def mouse_clicked():
    p5.fill(randint(30, 220), randint(30, 220), randint(30, 220), 100)

p5.run_sketch(setup=setup, draw=draw, mouse_clicked=mouse_clicked)